modules/ballerina-core/src/main/java/org/ballerinalang/model/types/BMapType.java

```diff
@@ -61,11 +70,35 @@ public BType getElementType() {
 
     @Override
     public TypeSignature getSig() {
-        return new TypeSignature(TypeSignature.SIG_REFTYPE, TypeEnum.MAP.getName());
+        return new TypeSignature(TypeSignature.SIG_MAP, constraint.getSig());
     }
 
     @Override
     public int getTag() {
         return TypeTags.MAP_TAG;
     }
+
+    @Override
+    public String toString() {
+        if (constraint == BTypes.typeAny) {
+            return super.toString();
```

Comment:
why do we need to special case any here? isn't else logic work for any as well?

Reply:
No it will not work because map constrained with any is internally how we maintain a map. For user it is just map. Cases where Eg:- logging errors etc. we use map instead even though we internally maintain a map of any. This is similar to constrained json.